contracts/semi-bridge/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+    #[cfg_attr(
+        feature = "std",
+        derive(scale_info::TypeInfo, ink_storage::traits::StorageLayout)
+    )]
+    struct Config {
+        rpc: String,
+        bridge_address: [u8; 20],
+    }
+
+    #[derive(Encode, Decode, Debug, PartialEq, Eq)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        BadOrigin,
+        NotConfigurated,
+        KeyRetired,
+        KeyNotRetiredYet,
+        UpstreamFailed,
+        BadAbi,
+        FailedToGetStorage,
+        FailedToDecodeStorage,
+        FailedToEstimateGas,
+        FailedToCreateExecutor,
+    }
+
+    type Result<T> = core::result::Result<T, Error>;
+
+    impl SemiBridge {
+        #[ink(constructor)]
+        pub fn default() -> Self {
+            Self {
+                owner: Self::env().caller(),
+                config: None,
+                key: Self::key_pair().private_key(),
+            }
+        }
+
+        /// Configures the bridge
+        #[ink(message)]
+        pub fn config(&mut self, rpc: String, bridge_address: H160) -> Result<()> {
+            self.ensure_owner()?;
+            self.config = Some(Config {
+                rpc,
+                bridge_address: bridge_address.into(),
+            });
+            Ok(())
+        }
+
+        /// Import a private key to override the interior account
+        #[ink(message)]
+        pub fn set_account(&mut self, private_key: Vec<u8>) -> H160 {
+            self.key = private_key.to_array();
+            self.wallet()
+        }
+
+        /// Returns the wallet address
+        #[ink(message)]
+        pub fn wallet(&self) -> H160 {
+            let keypair: KeyPair = self.key.into();
+            keypair.address()
+        }
+
+        /// Returns BadOrigin error if the caller is not the owner
+        fn ensure_owner(&self) -> Result<()> {
+            if self.env().caller() == self.owner {
+                Ok(())
+            } else {
+                Err(Error::BadOrigin)
+            }
+        }
+
+        /// Derives the key pair on the fly
+        fn key_pair() -> pink_web3::keys::pink::KeyPair {
+            pink_web3::keys::pink::KeyPair::derive_keypair(b"rollup-bridge")
+        }
+
+        /// Transfers tokens to the `address` derived from the contract's private key
+        ///
+        /// # Arguments
+        ///
+        /// * `src_chain`: an integer that represents the chain from which the asset is transferred
+        /// * `dest_chain`: the recipient of the tokens
+        /// * `token_rid`: token resource id
+        /// * `amount`: amount of token to be transferred
+        /// * `recipient`: the account that receives the tokens on Phala chain
+        #[ink(message)]
+        pub fn transfer(&self, token_rid: H256, amount: U256, recipient: H256) -> Result<()> {
+            let config = self
+                .config
+                .as_ref()
+                .map(Ok)
+                .unwrap_or(Err(Error::NotConfigurated))?;
```

Comment:
```suggestion
            let config = self
                .config
                .ok_or(Error::NotConfigurated)?;
```

Reply:
`as_ref()` is a must, otherwise `self.config` is moved from `self`, but I agree `ok_or` is better looking, let's add some rules to our style guide

```rust
let config = self
                .config.as_ref()
                .ok_or(Error::NotConfigurated)?;
```

Reply:
Let's keep `as_ref` then.